allennlp/predictors/predictor.py

```diff
@@ -58,6 +65,88 @@ def predict_json(self, inputs: JsonDict) -> JsonDict:
         instance = self._json_to_instance(inputs)
         return self.predict_instance(instance)
 
+    def json_to_labeled_instances(self, inputs: JsonDict) -> List[Instance]:
+        """
+        Converts incoming json to a :class:`~allennlp.data.instance.Instance`,
+        runs the model on the newly created instance, and adds labels to the
+        :class:`~allennlp.data.instance.Instance`s given by the model's output.
+        Returns
+        -------
+        List[instance]
+        A list of :class:`~allennlp.data.instance.Instance`
+        """
+        instance = self._json_to_instance(inputs)
+        outputs = self._model.forward_on_instance(instance)
+        new_instances = self.predictions_to_labeled_instances(instance, outputs)
+        return new_instances
+
+    def get_gradients(self,
+                      instances: List[Instance]) -> Tuple[Dict[str, Any], Dict[str, Any]]:
+        """
+        Gets the gradients of the loss with respect to the model inputs.
+
+        Parameters
+        ----------
+        instances: List[Instance]
+
+        Returns
+        -------
+        Tuple[Dict[str, Any], Dict[str, Any]]
+        The first item is a Dict of gradient entries for each input.
+        The keys have the form  ``{grad_input_1: ..., grad_input_2: ... }``
+        up to the number of inputs given. The second item is the model's output.
+
+        Notes
+        -----
+        Takes a ``JsonDict`` representing the inputs of the model and converts
+        them to :class:`~allennlp.data.instance.Instance`s, sends these through
+        the model :func:`forward` function after registering hooks on the embedding
+        layer of the model. Calls :func:`backward` on the loss and then removes the
+        hooks.
+        """
+        # clear embeddings and hooks
+        self.embedding_gradients = []
```

Comment:
You don't need to save these to `self` at all.  Just do what you do for the interpreters.  Make these local variables here:

```python
embedding_gradients = []
hooks = self._register_embedding_gradient_hooks(embedding_gradients)

...

for hook in hooks:
    hook.remove()

...
for index, grad in enumerate(embedding_gradients):
```

You don't need these variables anywhere outside of this method, so they should not be on `self`.